core/src/main/java/org/sonarsource/sonarlint/core/util/ws/BaseRequest.java

```diff
@@ -0,0 +1,171 @@
+    this.mediaType = s;
+    return (SELF) this;
+  }
+
+  public SELF setParam(String key, @Nullable String value) {
+    return setSingleValueParam(key, value);
+  }
+
+  public SELF setParam(String key, @Nullable Integer value) {
+    return setSingleValueParam(key, value);
+  }
+
+  public SELF setParam(String key, @Nullable Long value) {
+    return setSingleValueParam(key, value);
+  }
+
+  public SELF setParam(String key, @Nullable Float value) {
+    return setSingleValueParam(key, value);
+  }
+
+  public SELF setParam(String key, @Nullable Boolean value) {
+    return setSingleValueParam(key, value);
+  }
+
+  private SELF setSingleValueParam(String key, @Nullable Object value) {
+    checkArgument(!isNullOrEmpty(key), "a WS parameter key cannot be null");
+    if (value == null) {
+      return (SELF) this;
+    }
+    parameters.setValue(key, value.toString());
+
+    return (SELF) this;
+  }
+
+  public SELF setParam(String key, @Nullable Collection<? extends Object> values) {
+    checkArgument(!isNullOrEmpty(key), "a WS parameter key cannot be null");
+    if (values == null || values.isEmpty()) {
+      return (SELF) this;
+    }
+
+    parameters.setValues(key, values.stream()
+      .filter(Objects::nonNull)
+      .map(Object::toString)
+      .filter(value -> !value.isEmpty())
+      .collect(Collectors.toList()));
+
+    return (SELF) this;
+  }
+
+  @Override
+  public Map<String, String> getParams() {
+    return parameters.keyValues.keySet().stream()
+      .collect(Collectors.toMap(
+        Function.identity(),
+        key -> parameters.keyValues.get(key).get(0),
+        (v1, v2) -> {
+          throw new IllegalStateException(String.format("Duplicate key '%s' in request", v1));
+        },
+        LinkedHashMap::new));
+  }
+
+  @Override
+  public Parameters getParameters() {
+    return parameters;
+  }
+
+  private static class DefaultParameters implements Parameters {
+    // preserve insertion order
+    private final ListMultimap<String, String> keyValues = LinkedListMultimap.create();
+
+    @Override
+    @CheckForNull
+    public String getValue(String key) {
+      return keyValues.containsKey(key) ? keyValues.get(key).get(0) : null;
+    }
+
+    @Override
+    public List<String> getValues(String key) {
+      return keyValues.get(key);
+    }
+
+    @Override
+    public Set<String> getKeys() {
+      return keyValues.keySet();
+    }
+
+    private DefaultParameters setValue(String key, String value) {
+      checkArgument(!isNullOrEmpty(key));
+      checkArgument(value != null);
+
+      keyValues.putAll(key, singletonList(value));
+      return this;
+    }
+
+    private DefaultParameters setValues(String key, Collection<String> values) {
+      checkArgument(!isNullOrEmpty(key));
+      checkArgument(values != null && !values.isEmpty());
+
+      this.keyValues.putAll(key, values.stream().map(Object::toString).filter(Objects::nonNull).collect(Collectors.toList()));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) NullPointerException might be thrown as 'values' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259)